packages/block-editor/src/store/selectors.js

```diff
@@ -2941,3 +2941,11 @@ export const getBlockEditingMode = createRegistrySelector(
 			return parentMode === 'contentOnly' ? 'default' : parentMode;
 		}
 );
+
+export function getBlockBeingRenamed( state ) {
+	return state.blockBeingRenamed;
+}
+
+export function isBlockBeingRenamed( state, clientId ) {
+	return state?.blockBeingRenamed === clientId;
+}
```

Comment:
[As pointed out](https://github.com/WordPress/gutenberg/pull/53735#issuecomment-1685555721) by @andrewserong we may not need this global state. TBD.